### PR TITLE
Separate SDK Python tests in 3 suites and support running multiple suites/collections in one run

### PR DESCRIPTION
**Pull request by ccruzagralopes**
## What changed

- Separated the SDK Python tests in 3 suites:
   - Python Testing Suite: test cases that follow the template and need commissioning.
   - Python Testing Suite - No commissioning: test cases that follow the template and don't need commissioning.
   - Python Testing Suite - Legacy: test cases that don't follow the template so we don't know if they need commissioning or not.
- This change allows for running tests from different suites and collections in a single test run.

## New prompts

- "Make sure the DUT is in Commissioning Mode"
   - Shown at:
      - Setup for the "Python Testing Suite" suite.
      - Setup for each test case of the "Python Testing Suite - No commissioning" and the "Python Testing Suite - Legacy" suites.
Screenshot 2023-12-19 at 14 31 26

- "Should the DUT be commissioned to run this test case?"
   - Shown at:
      - Setup for each test case of the "Python Testing Suite - Legacy" suite.
Screenshot 2023-12-19 at 14 32 55

## Additional screenshots

- Tests from different test suites and collections running in the same execution:
Screenshot 2023-12-21 at 15 15 19

### CODE REVIEW

**Review by antonio-amjr — APPROVED** [review]

File: test_collections/sdk_tests/support/python_testing/models/python_test_parser.py [hunk 1]
```diff
@@ -163,6 +163,18 @@ def __parse_test_case(
     if pics_method:
         tc_pics = __retrieve_pics(pics_method)
 
+    # - PythonTestType.COMMISSIONING: test cases that have a commissioning first step
+    # - PythonTestType.NO_COMMISSIONING: test cases that follow the expected template
+    #   but don't have a commissioning first step
+    # - PythonTestType.LEGACY: test cases that don't follow the expected template
+    # We use the desc_[test_name] method as an indicator that the test case follows the
+    # expected template
+    python_type = PythonTestType.LEGACY
```
Comment (rquidute) [inline test_collections/sdk_tests/support/python_testing/models/python_test_parser.py:172]:
```suggestion
    python_test_type = PythonTestType.LEGACY
```
Maybe python_test_type has more meaning. What do you think?

**Review by rquidute — APPROVED** [review]

File: test_collections/sdk_tests/support/python_testing/models/python_test_parser.py (hunk 1 above)
Comment (ccruzagralopes) [reply test_collections/sdk_tests/support/python_testing/models/python_test_parser.py]: Done

File: test_collections/sdk_tests/support/python_testing/models/python_test_parser.py [hunk 2]
```diff
@@ -163,6 +163,18 @@ def __parse_test_case(
     if pics_method:
         tc_pics = __retrieve_pics(pics_method)
 
+    # - PythonTestType.COMMISSIONING: test cases that have a commissioning first step
+    # - PythonTestType.NO_COMMISSIONING: test cases that follow the expected template
+    #   but don't have a commissioning first step
+    # - PythonTestType.LEGACY: test cases that don't follow the expected template
```
Comment (hiltonlima) [inline test_collections/sdk_tests/support/python_testing/models/python_test_parser.py:169]: can we move these comments (or copy it) to class PythonTestType(Enum) definition?

**Review by hiltonlima — APPROVED** [review]

**Review by gladystonfranca — APPROVED** [review]

File: test_collections/sdk_tests/support/python_testing/models/python_test_parser.py (hunk 2 above)
Comment (ccruzagralopes) [reply test_collections/sdk_tests/support/python_testing/models/python_test_parser.py:169]: Done

**Pull request merged**